inc/vcf/record_cache.hpp

```diff
@@ -60,15 +78,44 @@ namespace ebi
          * Nonetheless, if the capacity is too small, it may cause incorrect reporting, such as reporting several times
          * the first occurrence or failing to report duplicates that are farther apart than the capacity.
          */
-        std::vector<std::unique_ptr<Error>> check_duplicates(const Record &record)
+        std::vector<std::unique_ptr<Error>> get_symbolic_duplicates()
+        {
+            return std::move(list_symbolic_duplicates);
+        }
+        
+
+        /**
+         * This function serves as the data generator to be used before utilizing get_duplicates() and get_symbolic_duplicates() functions.
+         *
+         * For a given Record, populates the list_duplicates and list_symbolic_duplicates vectors with
+         * non symbolic and symbolic duplicates respectively
+         *
+         * Even if more than one duplicate is found for the same variant during the file validation, each occurrence 
+         * is reported only once:
+         * - If just one processed record is equivalent to the parameter, both are reported in 2 different errors.
+         * - If more than one processed record is equivalent to the new one, only the new one is reported.
+         * 
+         * Nonetheless, if the capacity is too small, it may cause incorrect reporting, such as reporting several times
+         * the first occurrence or failing to report duplicates that are farther apart than the capacity.
+         */
+        void check_duplicates(const Record &record)
         {
             auto record_cores = normalize(record);
-            std::vector<std::unique_ptr<Error>> duplicates{};
 
+            list_duplicates.clear();
+            list_symbolic_duplicates.clear();
+            
             for (RecordCore &record_core: record_cores) {
-                std::pair<std::multiset<RecordCore>::iterator, std::multiset<RecordCore>::iterator> range =
-                        cache.equal_range(record_core);
 
+                // create references to the appropriate data structures for th alternate allele type
```

Comment:
I would be happy using `auto` in the next 3 statements, given they are assigning from attributes whose type we already now.